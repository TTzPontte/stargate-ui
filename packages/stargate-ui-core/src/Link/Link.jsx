import React from 'react';
import { useStyles } from '@pontte/stargate-ui-styles';
import clsx from 'clsx';

import Typography from '../Typography';

const styles = (theme) => {
  const { active, transition } = theme;

  const link = {
    textDecoration: 'none',
    color: 'inherit',
    transition: transition('opacity').ease,
    [active()]: {
      opacity: .6,
    },
  };

  return { link };
};

const Button = (props) => {
  const {
    className: inheritedClasses,
    ...typographyProps
  } = props;

  const [
    {
      link: classLink,
    },
  ] = useStyles(styles);

  return (
    <Typography
      element="a"
      variant="body"
      transform="uppercase"
      display="inline-flex"
      className={clsx(classLink, inheritedClasses)}
      {...typographyProps}
    />
  );
};

export default Button;
