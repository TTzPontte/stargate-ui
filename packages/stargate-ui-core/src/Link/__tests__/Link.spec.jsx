import React from 'react';

import Link from '../Link';

describe('Link', () => {
  it('should render properly', () => {
    const { container } = render(
      <Link>
        Click here
      </Link>,
    );

    expect(container).toMatchSnapshot();
  });
});
