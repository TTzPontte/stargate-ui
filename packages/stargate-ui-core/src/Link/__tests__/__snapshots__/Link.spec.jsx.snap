// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render properly 1`] = `
<div>
  <a
    class="stargate-ui-factory-0-2-11 stargate-ui-factory-d0-0-2-13 stargate-ui-typography-0-2-4 stargate-ui-typography-d0-0-2-8 stargate-ui-link-0-2-1 stargate-ui-linkUnderline-0-2-2 stargate-ui-linkUnderline-d0-0-2-3 stargate-ui-typographyFluid-0-2-7 stargate-ui-typographyFluid-d2-0-2-10"
  >
    Click here
  </a>
</div>
`;
